src/test/java/io/vertx/core/http/HttpTest.java

```diff
@@ -5992,7 +6005,29 @@ public void testHaProxyProtocolVersion2() throws Exception {
       .end();
     try {
       await();
-    }finally {
+    } finally {
+      proxy.stop();
+    }
+  }
+
+  @Test
+  public void testHAProxyProtocolIdleTimeout() throws Exception {
+    HAProxy proxy = new HAProxy(DEFAULT_HTTP_HOST, DEFAULT_HTTP_PORT, Buffer.buffer());
+    proxy.start(vertx);
+
+    server.close();
+    server = vertx.createHttpServer(createBaseServerOptions().
+      setIdleTimeout(2).
+      setSslHandshakeTimeout(2).
```

Comment:
why do we need `setSslHandshakeTimeout` here ? we don't need it for this test

Reply:
Not exactly. In HTTP2 the connection is closed not because of IdleHandler (Which is not yet added to the pipeline) but because of timeout waiting for handshake from (SSLHandler) which is configured using setSslHandshakeTimeout. 

This is how i found that the setting is not respected